frontend/lib/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -391,26 +391,39 @@ function DataFrame({
 
   // Determine if the table requires horizontal or vertical scrolling:
   React.useEffect(() => {
-    if (resizableContainerRef.current) {
-      // TODO(lukasmasuch): This is only a hacky and temporary solution until
-      // glide-data-grid provides a better way to determine this:
-      // https://github.com/glideapps/glide-data-grid/issues/784
-
-      // Get the bounds of the glide-data-grid scroll area (dvn-stack):
-      const scrollAreaBounds = resizableContainerRef.current
-        ?.querySelector(".dvn-stack")
-        ?.getBoundingClientRect()
-      if (scrollAreaBounds) {
-        setHasVerticalScroll(
-          scrollAreaBounds.height > resizableContainerRef.current.clientHeight
-        )
-        setHasHorizontalScroll(
-          scrollAreaBounds.width > resizableContainerRef.current.clientWidth
-        )
+    // The setTimeout is a workaround to get the scroll area bounding box
+    // after the grid has been rendered. Otherwise, the scroll area div
+    // (dvn-stack) might not have been created yet.
+    setTimeout(() => {
```

Comment:
Is `setTimeout` the right way of doing this? What happens if the timeout delay for rendering somehow past 1 second?

Reply:
It's 1 ms, not 1 second. We just need an extremely short delay to get the correct bounding box data. I don't really know another good way to get that done :(